crates/ruff_linter/src/rules/perflint/rules/manual_dict_comprehension.rs

```diff
@@ -139,8 +139,13 @@ pub(crate) fn manual_dict_comprehension(checker: &Checker, for_stmt: &ast::StmtF
             .semantic()
             .bindings
             .iter()
-            .find(|binding| target.range() == binding.range)
-            .expect("for-loop target binding must exist");
+            .find(|binding| target.range() == binding.range);
+        debug_assert!(target_binding.is_some());
```

Comment:
You could include your previous `expect` message as the `debug_assert` message, if you want.
```suggestion
        debug_assert!(target_binding.is_some(), "for-loop target binding must exist");
```